platform/autoupdate.services/src/org/netbeans/modules/autoupdate/updateprovider/AutoupdateCatalogParser.java

```diff
@@ -545,7 +545,11 @@ public void appendModuleAttributes (Attributes module) {
             String autoload = module.getValue (MODULE_ATTR_AUTOLOAD);
             String preferred = module.getValue(MODULE_ATTR_IS_PREFERRED_UPDATE);
                         
-            needsRestart = isFragment || needsrestart == null || needsrestart.trim ().length () == 0 ? null : Boolean.valueOf (needsrestart);
+            if (isFragment) {
+                needsRestart = true;
```

Comment:
Will that work with #715? The core idea is that we can install NbJavac (which includes a fragment) without restart!

Reply:
OK, let's take this change to #715. It just improves consistency with line ~570; I believe that  I added the isFragment condition erroneously here (parenthesis missing). Maybe better condition is needed: in fact restart is only needed if the hosting module is present/enabled prior to autoupdate operation.

Reply:
If "restart is only needed if the hosting module has already been present", then everything is OK.